OpenRA.Mods.Common/Traits/World/EditorResourceLayer.cs

```diff
@@ -188,37 +187,34 @@ public int CalculateRegionValue(CellRegion sourceRegion)
 			foreach (var cell in sourceRegion.CellCoords)
 			{
 				var mcell = cell.ToMPos(Map);
-				if (Map.Resources.Contains(mcell) && Map.Resources[mcell].Type != 0)
-				{
-					resourceValueInRegion++;
-					var rcell = Map.Resources[mcell];
-					if (ResourceTypesByIndex.TryGetValue(rcell.Type, out var resourceType) && resourceValues.TryGetValue(resourceType, out var resourceValuePerUnit))
-						resourceValueInRegion += Tiles[mcell].Density * resourceValuePerUnit;
-				}
+				byte resource;
+				if (Map.Resources.Contains(mcell)
```

Comment:
This is a monster of a line 😱

Can we please rewrite this with an `if (!Map.Resources.Contains(mcell)) continue;` instead of putting all of the logic in the one expression?

Reply:
sure